src/keydra/providers/splunk.py

```diff
@@ -234,20 +294,33 @@ def distribute(self, secret, destination):
     @classmethod
     def validate_spec(cls, spec):
 
-        for k, v in spec.items():
-            if not validators.length(k, min=2, max=75):
-                return False, 'Key {} failed length checks'.format(k)
+        if 'config' not in spec:
+            return (False, "Required section 'config' not present in spec")
 
-            # Don't key check lists, as it will check length of list not str
-            if (not isinstance(v, list) and
-                    not validators.length(v, min=2, max=75)):
-                return (False,
-                        'Value for key {} failed length checks'.format(k))
+        if 'hosts' not in spec['config']:
+            return (False, "Config must contain hosts")
 
-        if 'hosts' in spec:
-            for host in spec['hosts']:
-                if not validators.domain(host) and not validators.ipv4(host):
-                    return (False, 'Host {} not valid'.format(host))
+        for host in spec['config']['hosts']:
+            if not validators.domain(host) and not validators.ipv4(host):
+                return (False, 'Host {} not valid'.format(host))
+
+        if 'type' in spec['config']:
+            if spec['config']['type'] not in ['hectoken', 'user']:
+                return (
+                    False,
+                    'Invalid Splunk secret type'.format(
```

Comment:
It would be very user friendly to specify what the valid types are in the same message 🙂